app/controllers/api/journeys_controller.rb

```diff
@@ -85,6 +86,24 @@ def validate_params
       Journeys::ParamsValidator.new(data_params).validate!(action_name.to_sym)
     end
 
+    def validate_duplicate_journey
+      move_journeys = move.journeys
+      params_from_location = data_params['relationships']['from_location']['data']['id']
+      params_to_location = data_params['relationships']['to_location']['data']['id']
+
+      move_journeys.each do |journey|
+        next unless journey.state != 'cancelled' || journey.state != 'rejected' && journey.to_location_id == params_to_location && journey.from_location_id == params_from_location
```

Comment:
This line is really confusing to me, I think this would be more clear?
```suggestion
        next if journey.state == 'cancelled' || journey.state == 'rejected' || journey.to_location_id != params_to_location && journey.from_location_id != params_from_location
```